### PR TITLE
feat: add filterv2 functions

### PR DESCRIPTION
**Pull request by richard-ramos**
Adds filterV2 functions, and updates the API for relay since now it uses a content filter instead of only a pubsub topic

### CODE REVIEW

File: waku-bindings/src/node/legacyfilter.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+) -> Result<()> {
+    let filter_subscription_ptr = CString::new(
+        serde_json::to_string(filter_subscription)
+            .expect("FilterSubscription should always succeed to serialize"),
+    )
+    .expect("FilterSubscription should always be able to be serialized")
+    .into_raw();
+    let peer_id_ptr = CString::new(peer_id)
+        .expect("PeerId should always be able to be serialized")
+        .into_raw();
```
Comment (SionoiS) [inline waku-bindings/src/node/legacyfilter.rs:28]: Seams like `peer_id` could benefit from an implementation of `Into<CString>` or `TryInto<CString>`

Same with other struct that get changed into `CString`

**Review by SionoiS — APPROVED** [review]
FFI is not my forte but LGTM.

See my comments.

File: waku-bindings/src/general/mod.rs [hunk 2]
```diff
@@ -242,14 +242,103 @@ impl FilterSubscription {
...
+    }
+
+    pub fn content_topics(&self) -> &[WakuContentTopic] {
+        &self.content_topics
+    }
+
+    pub fn pubsub_topic(&self) -> Option<&WakuPubSubTopic> {
+        self.pubsub_topic.as_ref()
+    }
+}
```
[the remark this answers is not in the document]
Comment (richard-ramos) [reply waku-bindings/src/general/mod.rs:273]: Hm, I did this to keep it in the same style as the other structs that have been defined in waku-rust-bindings. I'll create an issue to remove the accessor methods everywhere!

[the remark this answers is not in the document]
Comment (richard-ramos) [reply waku-bindings/src/general/mod.rs:273]: Tracking this in https://github.com/waku-org/waku-rust-bindings/issues/82

File: waku-bindings/src/node/relay.rs [hunk 3]
```diff
@@ -191,55 +162,51 @@ pub fn waku_enough_peers(pubsub_topic: Option<WakuPubSubTopic>) -> Result<bool>
...
         );
 
-        drop(CString::from_raw(pubsub_topic_ptr));
+        drop(CString::from_raw(content_filter_ptr));
 
         out
     };
 
     handle_no_response(code, &error)
 }
```
Comment (SionoiS) [inline waku-bindings/src/node/relay.rs:189]: I feel like there must be a way to make this generic to avoid the repeated code. :thinking:

**Pull request merged**